### ISSUE TITLE
[BUG] slots detected as classes

### ISSUE THREAD

**Issue by BleedingDev** [francoismassart/eslint-plugin-tailwindcss#422]
**Describe the bug**
We use tailwind-variants with feature of slots. With beta version of plugin it incorrectly detects values there as invalid classes.

**To Reproduce**
1. Use tailwind-variants.
2. Define `compoundSlots` with `slots` and `class` keys inside the object.
3. Detect bad classes
4. Observe incorrect detection

**Expected behavior**
Detection only in classes, not somewhere else in tailwind-variants.

**Screenshots**

**Environment (please complete the following information):**
 - OS: Mac OS 26
 - Softwares + version used:
     -  VS Code + Terminal

**Additional context**
```ts
  compoundSlots: [
    {
      slots: ['autoplayTrigger', 'indicator', 'prevTrigger', 'nextTrigger'],
      class: [
        'p-carousel-trigger bg-carousel-trigger-bg',
        'focus:outline-none text-carousel-trigger-fg focus:ring-2 focus:ring-carousel-focus-ring focus:ring-offset-2',
      ],
    },
  ],
```
```bash
/libs/ui/src/molecules/carousel.tsx
  24:26  error  Classname 'scrollbar-hide' is not a Tailwind CSS class!   tailwindcss/no-custom-classname
  24:26  error  Classname 'autoplayTrigger' is not a Tailwind CSS class!  tailwindcss/no-custom-classname
  24:26  error  Classname 'indicator' is not a Tailwind CSS class!        tailwindcss/no-custom-classname
  24:26  error  Classname 'prevTrigger' is not a Tailwind CSS class!      tailwindcss/no-custom-classname
  24:26  error  Classname 'nextTrigger' is not a Tailwind CSS class!      tailwindcss/no-custom-classname
```

**eslint config file or live demo**
By providing a link to a live demo, a demo video or a github repo fixing the issue will be easier.

**Comment by amo12937** [comment]
## Workaround

Until this bug is fixed, you can work around it by adding `'slots'` to `ignoredKeys` in your ESLint configuration:

**For `.eslintrc` / `.eslintrc.js`:**
```json5
{
  settings: {
    tailwindcss: {
      ignoredKeys: ['compoundVariants', 'defaultVariants', 'slots']
    }
  }
}
```

**For `eslint.config.js` (flat config):**
```js
export default [
  ...tailwind.configs["flat/recommended"],
  {
    settings: {
      tailwindcss: {
        ignoredKeys: ['compoundVariants', 'defaultVariants', 'slots']
      }
    }
  }
];
```

## Proposed Fix

I would like to work on this fix.

I believe adding `'slots'` to the default `ignoredKeys` list would resolve this issue. Does this approach sound good?

An alternative would be to ignore `compoundSlots` entirely, but that would also skip validation of the `class` property within `compoundSlots`, which we probably want to check.
